cmd/kubeadm/app/master/manifests.go

```diff
@@ -96,10 +96,12 @@ func WriteStaticPodManifests(cfg *kubeadmapi.MasterConfiguration) error {
 			Env:           getProxyEnvVars(),
 		}, volumes...),
 		kubeScheduler: componentPod(api.Container{
-			Name:          kubeScheduler,
-			Image:         images.GetCoreImage(images.KubeSchedulerImage, cfg, kubeadmapi.GlobalEnvParams.HyperkubeImage),
-			Command:       getSchedulerCommand(cfg),
-			LivenessProbe: componentProbe(10251, "/healthz"),
+			Name:  kubeScheduler,
+			Image: images.GetCoreImage(images.KubeSchedulerImage, cfg, kubeadmapi.GlobalEnvParams.HyperkubeImage),
+			// TODO: Using non-standard port here so self-hosted scheduler can come up:
+			// Use the regular port if this is not going to be a self-hosted deployment.
+			Command:       getSchedulerCommand(cfg, 10260),
```

Comment:
In the case of using CNI:

1. kubelet reaches temp-static-apiserver
1. kubelet sees that it has a controller-manager pod scheduled to it (via deployment) (note: cm does not use host-networking)
1. kubelet won't start controller-manager pod yet  because there is no CNI config / no pod network.
1. At some point in bootstrap the CNI plugin pod is deployed (via daemonset). This does use host-networking so it is able to start.
1. Now kubelet can start pod-network pods, and starts the controller-manager
1. Controller-manager actually doesn't do anything yet because it's leader-elected and the temporary static control-plane is still running.
1. Once all control-plane replacement pods are running, pivot occurs the self-hosted controller-manager takes over (and there shouldn't be any port conflicts).